>
--- FAIL: TestDescribeContainers (0.00s)
	describe_test.go:305: Test case 8: expected to find "cpu" in output: "Containers:\n  test:\n    Container ID:\t\n    Image:\timage\n    Image ID:\t\n    Port:\t\n    State:\tWaiting\n    Ready:\tTrue\n    Restart Count:\t7\n    Environment Variables:\t<none>\n"
	describe_test.go:305: Test case 8: expected to find "BestEffort" in output: "Containers:\n  test:\n    Container ID:\t\n    Image:\timage\n    Image ID:\t\n    Port:\t\n    State:\tWaiting\n    Ready:\tTrue\n    Restart Count:\t7\n    Environment Variables:\t<none>\n"
	describe_test.go:305: Test case 8: expected to find "memory" in output: "Containers:\n  test:\n    Container ID:\t\n    Image:\timage\n    Image ID:\t\n    Port:\t\n    State:\tWaiting\n    Ready:\tTrue\n    Restart Count:\t7\n    Environment Variables:\t<none>\n"
	describe_test.go:305: Test case 8: expected to find "BestEffort" in output: "Containers:\n  test:\n    Container ID:\t\n    Image:\timage\n    Image ID:\t\n    Port:\t\n    State:\tWaiting\n    Ready:\tTrue\n    Restart Count:\t7\n    Environment Variables:\t<none>\n"

Looks like there are still some issues with the unit tests

--- a/pkg/kubectl/describe.go
+++ b/pkg/kubectl/describe.go
@@ -540,6 +540,7 @@ func describePod(pod *api.Pod, events *api.EventList) (string, error) {
 			}
 		}
 		describeVolumes(pod.Spec.Volumes, out, "")
+		fmt.Fprintf(out, "QoS Tier:\t%s\n", qosutil.GetPodQos(pod))
 		if events != nil {
 			DescribeEvents(events, out)
 		}
@@ -841,15 +842,6 @@ func describeContainers(label string, containers []api.Container, containerStatu
 			for _, arg := range container.Args {
 				fmt.Fprintf(out, "      %s\n", arg)
 			}
-		}
-
-		resourceToQoS := qosutil.GetQoS(&container)
-		if len(resourceToQoS) > 0 {
-			fmt.Fprintf(out, "    QoS Tier:\n")
-		}
-		for _, resource := range SortedQoSResourceNames(resourceToQoS) {
-			qos := resourceToQoS[resource]
-			fmt.Fprintf(out, "      %s:\t%s\n", resource, qos)
 		}
 
 		resources := container.Resources

--- a/pkg/kubectl/describe_test.go
+++ b/pkg/kubectl/describe_test.go
@@ -238,19 +238,6 @@ func TestDescribeContainers(t *testing.T) {
 			},
 			expectedElements: []string{"test", "State", "Waiting", "Ready", "True", "Restart Count", "7", "Image", "image", "time", "1000"},
 		},
-		// QoS classes
-		{
-			container: api.Container{
-				Name:  "test",
-				Image: "image",
-			},
-			status: api.ContainerStatus{
-				Name:         "test",
-				Ready:        true,
-				RestartCount: 7,
-			},
-			expectedElements: []string{"cpu", "BestEffort", "memory", "BestEffort"},
-		},
 		// Using limits.
 		{
 			container: api.Container{

--- a/test/e2e/kubectl.go
+++ b/test/e2e/kubectl.go
@@ -508,8 +508,8 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 					{"IP:"},
 					{"Controllers:", "ReplicationController/redis-master"},
 					{"Image:", redisImage},
-					{"cpu:", "BestEffort"},
 					{"State:", "Running"},
+					{"QoS Tier:", "BestEffort"},
 				}
 				checkOutput(output, requiredStrings)
 			})

